network/gossip.go

```diff
@@ -53,32 +58,62 @@ func (t *Topic) Subscribe(handler func(obj interface{})) error {
 	return nil
 }
 
+func (t *Topic) Close() error {
+	close(t.unsubscribeCh)
+	t.wg.Wait()
```

Comment:
Is the `waitGroup` really necessary? I would recommend remove this `wg` field.

Reply:
wait pubsub send an unsubscribe announcement to the network. 
[network/gossip.go#L84](https://github.com/dogechain-lab/dogechain/blob/fcfcbc440426e08bff34b90d851de2b0928d33cb/network/gossip.go#L84)

Reply:
maybe need time out handle